lib/Specio/Exception.pm

```diff
@@ -29,6 +29,9 @@ use Specio::OO;
         stack_trace => {
             init_arg => undef,
         },
+        context => {
+            isa      => 'HashRef',
```

Comment:
I wonder if this really needs to be a HashRef. Is there any reason not to allow any defined value? I guess this would make including it in the `as_string` output harder, but as it stands this output won't be very helpful if the hashref contains any references of its own.